pmd-core/src/main/java/net/sourceforge/pmd/processor/AbstractPMDProcessor.java

```diff
@@ -108,25 +115,44 @@ protected RuleSets createRuleSets(RuleSetFactory factory, Report report) {
     @SuppressWarnings("PMD.CloseResource")
     // the data sources must only be closed after the threads are finished
     // this is done manually without a try-with-resources
+    @Deprecated
     public void processFiles(RuleSetFactory ruleSetFactory, List<DataSource> files, RuleContext ctx,
-            List<Renderer> renderers) {
+                             List<Renderer> renderers) {
         try {
             final RuleSets rs = createRuleSets(ruleSetFactory, ctx.getReport());
-            configuration.getAnalysisCache().checkValidity(rs, configuration.getClassLoader());
+            processFiles(rs, files, ctx, renderers);
+        } finally {
```

Comment:
I think, we don't need the try-finally here - the files are opened/closed in the other processFiles method